lib/buckit/cli/create.rb

```diff
@@ -0,0 +1,92 @@
+require "buckit/cli/base_cmd"
+
+module Buckit
+  module CLI
+
+    class Create < BaseCmd
+      attr_accessor :region
+
+      def initialize
+        super 'create', false, false
+
+        @short_desc = "Create a new bucket"
+
+        self.options = CmdParse::OptionParserWrapper.new do |opt|
+          opt.on("-r", "--region REGION", "specify a region, default eu-west-1") {|region|
+            @region = region
+          }
+        end
+      end
+
+      def run s3, bucket, args
+        raise WrongUsage.new(nil, "You need to supply a bucket name") if not bucket
+
+        # Create bucket
+        create_bucket(s3, bucket)
+        credentials = create_user(bucket)
+        create_policy(bucket)
+        add_cors(s3, bucket)
+
+        puts "FOG_DIRECTORY=#{bucket} FOG_PROVIDER=AWS FOG_REGION=#{@region} ASSET_HOST=#{Aws::S3::Resource.new.bucket(bucket).url} AWS_ACCESS_KEY_ID=#{credentials.access_key_id} AWS_SECRET_ACCESS_KEY=#{credentials.secret_access_key}"
+      end
+
+      protected
+      def create_bucket s3, bucket
+        s3.create_bucket(
+          acl: "public-read",
+          bucket: bucket
+        )
+      end
+
+      def create_user bucket
+        begin
+          iam = Aws::IAM::Client.new
+          iam.create_user(user_name: bucket, path: '/apps/')
+          credentials = iam.create_access_key(user_name: bucket)
+          credentials.first.access_key
+        rescue Aws::IAM::Errors::ServiceError
+          raise FailureFeedback.new("Failed creating IAM")
+        end
+      end
+
+      def create_policy bucket
```

Comment:
Misleading name, this method doesn't just create a policy, it creates a policy and applies it to a user, which is implicitly tied to the bucket name